scripts/ci/check-lockbud.sh

```diff
@@ -0,0 +1,17 @@
+#!/bin/bash
+
+# Run lockbud to check for deadlocks and capture the output
+output=$(cargo lockbud -k deadlock -b -l tokio_util 2>&1)
+
+# Check if lockbud returned any issues
+if echo "$output" | grep -q '"bug_kind"'; then
```

Comment:
Note: this may produce false negatives if `cargo lockbud` failed for other reasons e.g. 
1. compilation error 
2. lockbud internal issues 

Although i think this is already better than before! Given that (1) compilation errors should fail other jobs as well, and we haven't seen (2) happened before (it may be very unlikely), I think we can go ahead and merge this.

Reply:
Agree. 
This script was only intended to be a quick fix.

I do suggest we 
- centralize this script as currently its copy pasta between anchor and LH.
- ultimately improve the lockbud cli so it can return and error code with the result.